tests/framework/e2e/cluster.go

```diff
@@ -640,6 +636,12 @@ func (cfg *EtcdProcessClusterConfig) EtcdServerProcessConfig(tb testing.TB, i in
 		if flag == "experimental-snapshot-catchup-entries" && !CouldSetSnapshotCatchupEntries(execPath) {
 			continue
 		}
+		if flag == "experimental-compaction-batch-limit" {
+			continue
+		}
+		if flag == "compaction-batch-limit" {
+			flag = "experimental-compaction-batch-limit"
+		}
```

Comment:
```suggestion
		// To be compatible with 3.4 & 3.5, as they don't have flag `--compaction-batch-limit`.
		if flag == "compaction-batch-limit" {
			flag = "experimental-compaction-batch-limit"
		}
```

But it's still a little tricky. Starting from 3.7, we will remove the experimental flags. So 
- 3.4 & 3.5 only have the experimental flags
- 3.6 have both flags
- 3.7 will only have the non-experimental flags.

So we may need to check the binary version and decide to use which flag. It's better to get them wrapped in a separate function.

Reply:
The upgrade/downgrade robustness test can set this config, but will not update the flags before restart. We would need to rewrite the e2e framework to reload flags from config on every restart.

Reply:
> We would need to rewrite the e2e framework to reload flags from config on every restart.

It seems that we need to reconcile all the config items once we update the binary (i.e. from 3.6 to 3.5, or from 3.5 to 3.6)

Reply:
use the `experimentalNonBoolFlagMigrationMap` to map the flags in migration for now? It should work before we rewrite the framework functions.

Reply:
With this PR I will try to address the main issue of robustness test failing.

Reply:
The only concern is that `--compaction-batch-limit` won't be covered by the e2e test anymore, because you always replace it with the `--experimental-compaction-batch-limit`.

Are you going to resolve it in a separate followup PR?

Reply:
It was and it is, but it won't if we merge this PR.

Reply:
I will let this PR in if you are aware of it and are planing to resolve it in a followup PR.

Reply:
Let me simplify the it further to not require a followup.